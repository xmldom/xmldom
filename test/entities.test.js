'use strict';

const { describe, expect, test } = require('@jest/globals');
const { getTestParser } = require('./get-test-parser');
const { MIME_TYPE } = require('../lib/conventions');
const { HTML_ENTITIES, XML_ENTITIES } = require('../lib/entities');

describe('XML_ENTITIES', () => {
	test('should not have a prototype', () => {
		expect(XML_ENTITIES).not.toHaveProperty('prototype');
		expect(XML_ENTITIES).not.toHaveProperty('__proto__');
	});
});
describe('HTML_ENTITIES', () => {
	test('should not have a prototype', () => {
		expect(HTML_ENTITIES).not.toHaveProperty('prototype');
		expect(HTML_ENTITIES).not.toHaveProperty('__proto__');
	});
});
describe('entity replacement ignores js prototype chain', () => {
	test("should not pick up 'entities' from the prototype chain", () => {
		const source = `
		<xml>
			<hasOwnProperty>&hasOwnProperty;</hasOwnProperty> 
			<proto>&__proto__;</proto> 
			<constructor>&constructor;</constructor>
		</xml>
`;
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString(source, MIME_TYPE.XML_TEXT).toString();

		expect({ actual, ...(errors.length ? { errors } : undefined) }).toMatchSnapshot();
	});
});
