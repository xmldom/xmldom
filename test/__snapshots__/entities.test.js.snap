// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entity replacement ignores js prototype chain should not pick up 'entities' from the prototype chain 1`] = `
{
  "actual": "
		<xml>
			<hasOwnProperty>&amp;hasOwnProperty;</hasOwnProperty> 
			<proto>&amp;__proto__;</proto> 
			<constructor>&amp;constructor;</constructor>
		</xml>",
  "errors": [
    [
      "error",
      "entity not found:&hasOwnProperty;",
      {
        "columnNumber": 44,
        "lineNumber": 5,
      },
    ],
    [
      "error",
      "entity not found:&__proto__;",
      {
        "columnNumber": 44,
        "lineNumber": 5,
      },
    ],
    [
      "error",
      "entity not found:&constructor;",
      {
        "columnNumber": 44,
        "lineNumber": 5,
      },
    ],
  ],
}
`;
